gtk4/src/pad_action_entry.rs

```diff
@@ -21,32 +27,68 @@ impl PadActionEntry {
         action_name: &str,
     ) -> Self {
         assert_initialized_main_thread!();
-        Self {
-            type_,
-            index,
-            mode,
-            label: label.to_owned(),
-            action_name: action_name.to_owned(),
+        unsafe {
+            Self::unsafe_from(ffi::GtkPadActionEntry {
+                type_: type_.into_glib(),
+                index,
+                mode,
+                label: label.to_glib_full(),
+                action_name: action_name.to_glib_full(),
+            })
         }
     }
 
     pub fn type_(&self) -> PadActionType {
-        self.type_
+        unsafe { from_glib(self.inner.type_) }
     }
 
     pub fn index(&self) -> i32 {
-        self.index
+        self.inner.index
     }
 
     pub fn mode(&self) -> i32 {
-        self.mode
+        self.inner.mode
     }
 
     pub fn label(&self) -> &str {
-        &self.label
+        unsafe { CStr::from_ptr(self.inner.label).to_str().unwrap() }
     }
 
     pub fn action_name(&self) -> &str {
-        &self.action_name
+        unsafe { CStr::from_ptr(self.inner.action_name).to_str().unwrap() }
     }
 }
+
+unsafe fn copy_action_entry(
+    action_entry: *const ffi::GtkPadActionEntry,
+) -> *mut ffi::GtkPadActionEntry {
+    let copy = glib::ffi::g_malloc0(std::mem::size_of::<ffi::GtkPadActionEntry>())
+        as *mut ffi::GtkPadActionEntry;
+    copy_into_action_entry(copy, action_entry);
+    copy
+}
+
+unsafe fn free_action_entry(action_entry: *mut ffi::GtkPadActionEntry) {
+    clear_action_entry(action_entry)
+}
+
+unsafe fn init_action_entry(action_entry: *mut ffi::GtkPadActionEntry) {
+    std::ptr::write(action_entry, std::mem::zeroed());
```

Comment:
That's the default already, zero initialization

Reply:
if i provide a copy_into/clear an init function is required

Reply:
That's strange. This https://github.com/gtk-rs/gtk-rs-core/blob/2e2af7ceaf2ec0e4eee474588f0fd0da86fad863/glib/src/boxed_inline.rs#L73 here should allow for exactly your use case

Reply:
the problem is more here i believe https://github.com/gtk-rs/gtk-rs-core/blob/2e2af7ceaf2ec0e4eee474588f0fd0da86fad863/glib/src/wrapper.rs#L308-L312

Reply:
That should allow passing in only those though, and not the others.

Reply:
What exactly is the problem? Your current version that does not provide copy/free compiles fine for me.

Reply:
Yes, it is more the init function that you cannot drop